### PR TITLE
Fix: terra.pfb manifest IT fails on prod (#3138)

### PR DESCRIPTION
**Pull request by jessebrennan**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in sandbox and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Comment by jessebrennan** [comment]
I have a question about how to reduce redundancy with the test and if it's worth keeping on that I will bring up in PL.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: test/indexer/data/587d74b4-1075-4bbf-b96a-4d1ede0481b2.metadata.json [hunk 1]
```diff
@@ -1148,62 +1184,8 @@
...
-                    {
-                        "protocol_type": "enrichment_protocol",
-                        "protocol_id": "d7820ce6-f0f4-4778-9353-423626dbd7c0"
-                    }
-                ]
-            },
-            {
-                "process": "908d468a-eb62-4d19-8706-c5f708a14940",
-                "inputs": [
-                    "242e38d2-c975-47ee-800a-6645b47e92d2"
```
Comment (hannes-ucsc) [inline test/indexer/data/587d74b4-1075-4bbf-b96a-4d1ede0481b2.metadata.json:1206]: I mean this big deletion.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Would also like to see evidence that the added/changed test reproduces the issue.

File: test/indexer/data/587d74b4-1075-4bbf-b96a-4d1ede0481b2.metadata.json [hunk 2]
```diff
@@ -1148,62 +1184,8 @@
             {
                 "process": "908d468a-eb62-4d19-8706-c5f708a14940",
                 "inputs": [
-                    "242e38d2-c975-47ee-800a-6645b47e92d2"
```
[the remark this answers is not in the document]
Comment (jessebrennan) [reply test/indexer/data/587d74b4-1075-4bbf-b96a-4d1ede0481b2.metadata.json:1151]: Did you read the commit message?

**Comment by jessebrennan** [comment]
> Would also like to see evidence that the added/changed test reproduces the issue.

The evidence is there in the history. I can explain it in parking lot.

**Comment by jessebrennan** [comment]
I will redo this with the following commits:

- Use newer bundle for PFB manifest test
- Revert hotfix
- Modify bundle (dynamically) to expose error from prod
- Fix

**Comment by jessebrennan** [comment]
@hannes-ucsc I'm having trouble finding a new bundle that contains zarrs. Having zarrs is useful for coverage, and to make sure "related_files" make it into the final manifest. There are (as best I can tell) no zarrs in DCP2. The most recent DCP1 zarr I can find is from September 2019. Is this recent enough or should I drop the zarr requirement?

I will bring this up in PL

File: test/service/test_manifest.py [hunk 3]
```diff
@@ -203,6 +204,61 @@ def test_pfb_manifest(self):
                         with open(results_file, 'w') as f:
                             json.dump(records, f, indent=4, sort_keys=True)
 
+    def _add_ageless_donor(self, new_bundle):
+        """
+        We add a new donor which lacks "age" metadata to see if PFB generation
+        can handle aggregating both kinds of donors.
+        """
+        new_bundle = self._load_canned_bundle(new_bundle)
+        new_bundle.metadata_files['donor_organism_1.json'] = {
```
Comment (hannes-ucsc) [inline test/service/test_manifest.py:213]: No literal, duplicate the first donor programmatically instead.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Please squash and push commits individually. No bump.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**